job_service/adapter/local_storage.py

```diff
@@ -58,3 +59,9 @@ def get_importable_datasets() -> List[ImportableDataset]:
     if ARCHIVE_DIR.exists():
         datasets += get_datasets_in_directory(ARCHIVE_DIR, is_archived=True)
     return datasets
+
+def delete_importable_datasets(dataset_name):
+    try:
+        os.remove(f"{INPUT_DIR}/{dataset_name}")
+    except (FileNotFoundError, OSError) as e:
+        raise NotFoundException(f"File {dataset_name} not found: {str(e)}")
```

Comment:
It would be better to change this to 
```py
raise NotFoundException(f"File {dataset_name} not found") from e
```
This prevents us from sending information about the file system to the client when we catch the error in app.py, but still logs that info 👍🏻